translator/translate/logs/ruleConcurrency.go

```diff
@@ -3,25 +3,83 @@
 
 package logs
 
-import "github.com/aws/amazon-cloudwatch-agent/translator"
+import (
+	"errors"
+	"runtime"
 
-const ConcurrencySectionKey = "concurrency"
+	"github.com/aws/amazon-cloudwatch-agent/translator"
+	"github.com/aws/amazon-cloudwatch-agent/translator/context"
+	"github.com/aws/amazon-cloudwatch-agent/translator/translate/logs/constants"
+	"github.com/aws/amazon-cloudwatch-agent/translator/translate/util"
+)
+
+const (
+	ConcurrencySectionKey = "concurrency"
+)
+
+var (
+	logFileCollectListPath      = util.Path(constants.SectionKeyLogsCollected, constants.SectionKeyFiles, constants.SectionKeyCollectList)
+	checkTimestampFormatVisitor = util.NewSliceVisitor(util.NewVisitor(isMissingTimestampFormat))
+	// based on AWS Java SDK default
+	defaultConcurrency = max(runtime.NumCPU(), 8)
+)
 
 type Concurrency struct {
 }
 
 func (c *Concurrency) ApplyRule(input any) (string, any) {
-	result := map[string]interface{}{}
-	_, val := translator.DefaultCase(ConcurrencySectionKey, float64(0), input)
+	result := map[string]any{}
+	_, val := translator.DefaultCase(ConcurrencySectionKey, nil, input)
 	var concurrency int
-	if v, ok := val.(float64); ok && v > 1 {
+	if v, ok := val.(float64); ok {
 		concurrency = int(v)
+	} else {
+		concurrency = determineDefault(input)
+	}
+	if concurrency > 1 {
 		result[ConcurrencySectionKey] = concurrency
+		GlobalLogConfig.Concurrency = concurrency
+	} else {
+		GlobalLogConfig.Concurrency = -1
```

Comment:
Whats the difference between 0 and -1

Reply:
No difference. Anything 1 or lower disables concurrency.

Reply:
nit: Can we have a constant called DISABLE_CONCURRENCY=-1, and keep it consistent with that